File: homework_js/homework_js/Form1.js
// Importa le classi Adversary e Histogram
import {Adversary} from './Adversary.js';
import {Histogram} from './Histogram.js';
import {LineChart} from './LineChart.js';



    function GraphHAndler(){
        this.mouseDownLocation = { X: 0, Y: 0 };
        this.isDragging = false;
        this.graphBitmap = null;

        // Genera gli attacchi
        let adv = new Adversary(1000, 20, 0.5);
        adv.generateAttacks();

        // Recupera i dati
        let data1 = adv.getLineChart1AttackList();
        let data2 = adv.getLineChart2AttackList();
        let data3 = adv.getLineChart3AttackList();
        let data4 = adv.getLineChart4AttackList();

        let chart1 = new LineChart(pictureBox1);
        chart1.drawChart(data1);

        let chart2 = new LineChart(pictureBox2);
        chart2.drawChart(data2);

        let chart3 = new LineChart(pictureBox3);
        chart3.drawChart(data3);

        let chart4 = new LineChart(pictureBox4);
        chart4.drawChart(data4);

        let k = 20;
        let histogramDistChart1 = adv.createHistoDistrib(data1, k);
        let histogramDistChart2 = adv.createHistoDistrib(data2, k);
        let histogramDistChart3 = adv.createHistoDistrib(data3, k);
        let histogramDistChart4 = adv.createHistoDistrib(data4, k);

        let histogram1 = new Histogram(this, chart1, histogramDistChart1, k);
        let histogram2 = new Histogram(this, chart2, histogramDistChart2, k);
        let histogram3 = new Histogram(this, chart3, histogramDistChart3, k);
        let histogram4 = new Histogram(this, chart4, histogramDistChart4, k);
    }

    Form1_Load(sender, e) {
        // Codice da eseguire durante il caricamento del form
    }
}

File: homework3_Csh/homework3_Csh/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace homework3_Csh
{
    public partial class Form1 : Form
    {
        private Point mouseDownLocation;
        private bool isDragging = false;


        private Bitmap graphBitmap;

        public Form1()
        {
            InitializeComponent();

            // generate attacks
            Adversary adv = new Adversary(1000, 20, 0.5f);
            adv.generateAttacks();
            // retrieve data
            List<List<int>> data1 = adv.GetLineChart1AttackList();
            List<List<int>> data2 = adv.GetLineChart2AttackList();
            
            List<List<float>> data3 = adv.GetLineChart3AttackList();
            List<List<float>> data4 = adv.GetLineChart4AttackList();
            
            // generate LineCharts
            LineChart chart1 = new LineChart(this, pictureBox1);
            chart1.DrawChart(this, data1);

            LineChart chart2 = new LineChart(this, pictureBox2);
            chart2.DrawChart(this, data2);

            LineChart chart3 = new LineChart(this, pictureBox3);
            chart3.DrawChart(this, data3);

            LineChart chart4 = new LineChart(this, pictureBox4);
            chart4.DrawChart(this, data4);

            int k = 20;

            // retrieve histogram data
            Dictionary<int, int> histogramDistChart1 = adv.createHistoDistrib(data1, k);
            Dictionary<int, int> histogramDistChart2 = adv.createHistoDistrib(data2, k);
            Dictionary<int, float> histogramDistChart3 = adv.createHistoDistrib(data3, k);
            Dictionary<int, float> histogramDistChart4 = adv.createHistoDistrib(data4, k);

            // generate Histograms
            Histogram histogram1 = new Histogram(this, chart1, histogramDistChart1, k);
            Histogram histogram2 = new Histogram(this, chart2, histogramDistChart2, k);
            Histogram histogram3 = new Histogram(this, chart3, histogramDistChart3, k);
            Histogram histogram4 = new Histogram(this, chart4, histogramDistChart4, k);
            
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            
        }
    }
}
